File: Transform to prime[08-December-2023]/Solution.cpp
//   >> CPP CODE <<

class Solution {
public:
    int MAX;
    bool primes[1000001];
    Solution() {
        MAX = 1000001;
        memset(primes, true, sizeof(primes));
        primes[1] = false;

        for (int i = 2; i*i < MAX; i++) {
            if (primes[i]) {
                for (int j = i * 2; j < MAX; j += i) {
                    primes[j] = false;
                }
            }
        }
    }
    int minNumber(int arr[], int N) {
        int sum = 0;

        for (int i = 0; i < N; i++) sum += arr[i];

        int st = sum;

        while (primes[st] == false)
            st++;

        return st - sum;
    }
};

File: Transform to prime[08-December-2023]/Solution.py
#    >> PYTHON CODE <<

MAX=1000001
primes=[True]*MAX
primes[1]=False
for i in range(2,int(MAX**(0.5))+1):
    if(primes[i]):
        for j in range(2*i,MAX,i):
            primes[j]=False

class Solution:
            
    
    def minNumber(self, arr,n):

        sm=sum(arr)

        st=sm
        
        while(primes[st]==False): st+=1
        
        return int(st-sm)
